### PR TITLE
Ignore prototypes in object iteration

### PR DESCRIPTION
**Pull request by jharmn**
When prototype methods have been added to the GDD schema, iterations of object properties will fail.

http://yuiblog.com/blog/2006/09/26/for-in-intrigue/

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -297,7 +305,7 @@ function processParameterList(method) {
 
 function processParameter(name, param) {
   assert.ok(!('$ref' in param), 'parameter cannot contain $ref: '+param);
-  assert.ok(['query', 'path'].indexOf(param.location) > -1, 'parameter type must be \'query\' or \'path\'');
+  assert.ok(['query', 'path'].indexOf(param.location) > -1, 'parameter location must be \'query\' or \'path\'');
```
Comment (jasonh-n-austin) [inline src/index.js]: Also tweaked this, as I realized the message was inaccurate from my last PR.

**Comment by IvanGoncharov** [comment]
@jasonh-n-austin Can you use [forOwn](https://lodash.com/docs#forOwn) function?
Everything else, looks good :+1:

**Comment by jasonh-n-austin** [comment]
I tried [forOwn](https://lodash.com/docs#forOwn), but it doesn't seem to work the same way. It still iterates prototyped methods.

**Comment by IvanGoncharov** [comment]
What is your environment?
I tried lodash example and it works fine.  Can you try it on your system?
```js
function Foo() {
  this.a = 1;
  this.b = 2;
}

Foo.prototype.c = 3;

_.forOwn(new Foo, function(value, key) {
  console.log(key);
});
// → logs 'a' and 'b' (iteration order is not guaranteed)
```
I also inspect lodash implementation and it use standard `Object.keys`, can you log output of this function?

**Comment by IvanGoncharov** [comment]
@jasonh-n-austin Did you had a chance to look at it?

**Comment by IvanGoncharov** [comment]
Hi @jasonh-n-austin,

Happy New Year :christmas_tree: :santa: 
Is this issue still valid?

**Comment by jasonh-n-austin** [comment]
Totally forgot about this. I'll take a look at it by the end of the week, thanks for the reminder.

**Comment by jasonh-n-austin** [comment]
Ran your test script on `v0.10.25`/Ubuntu and `v0.12.1`/Mac, both printed `a`,`b`. 

I'll re-run this with my test data and see if I can figure out what scenario made this necessary.

**Comment by jasonh-n-austin** [comment]
FYI @crankydillo has been working on some improvements (which are now included in this PR). I've also asked for his help to pin down why we needed this prototype exclusion to make things work (my brain is already full of cobwebs about it).

File: test/unit/test-index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+								'\'number\', \'integer\', \'boolean\'; was \'foo\'');
+	  }
+	});
+
+  });
+
+});
+
+function mkGdd() {
+  return {
```
Comment (IvanGoncharov) [inline test/unit/test-index.js]: Can you please put it in separate file and just do something like this:
```json
var sample = require('./google-discovery-sample.json')
```

File: package.json [hunk 3]
```diff
@@ -30,6 +32,8 @@
     "traverse": "~0.6.6"
   },
   "devDependencies": {
+	"chai": "*",
+	"mocha": "*",
```
Comment (IvanGoncharov) [inline package.json]: indent

**Comment by IvanGoncharov** [comment]
@jasonh-n-austin  
>why we needed this prototype exclusion to make things work 
I understand why. But I'm don't understand why `forOwn` not working?

can you try to do:
```
data = JSON.parse(JSON.stringify(data));
```
before passing it into the lib?

**Comment by MikeRalphson** [comment]
Assuming this PR is no longer relevant/current? Happy to re-open if necessary.

**Pull request closed without merging**